### PR TITLE
#160451460 Users should be able to rate articles

### PR DESCRIPTION
**Pull request by samdiano**
What does this PR do?
This PR adds a star rating to an article

Description of Task to be completed?
- Display rating average with 5 stars
- Add rating through the stars displayed
#### How should this be manually tested?
Go to the et single article url, you should see a page showing an article with it's rating average
#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?
#160451460

### CODE REVIEW

File: src/requests/ArticleRequests.js [hunk 1]
```diff
@@ -30,6 +36,22 @@ export const viewArticle = payload => (dispatch) => {
...
+  const headers = {
+    'Content-Type': 'application/json;charset=UTF-8',
+  };
+  dispatch(asyncActions(RATE_ARTICLE).loading(true));
+  axios.post(`${articleConstant.RATE_ARTICLE_URL}/${slug}/rate`, { rating }, headers)
+    .then((response) => {
+      if (response.status === 200) {
+        dispatch(asyncActions(RATE_ARTICLE).success(response.data.article));
+        dispatch(asyncActions(RATE_ARTICLE).loading(false));
+        // dispatch(msgInfoActions.success([response.data.message]));
```
Comment (s-akinrele) [inline src/requests/ArticleRequests.js]: remove commented code

Comment (samdiano) [reply src/requests/ArticleRequests.js]: fixed

File: .env [hunk 2]
```diff
@@ -0,0 +1,4 @@
+PORT=8080
```
Comment (s-akinrele) [inline .env]: @samdiano this should be in a git ignore file

File: src/constants/Constants.js [hunk 3]
```diff
@@ -1,9 +1,10 @@
-const { BASE_URL } = process.env;
+const BASE_URL = 'https://phlox-ah-staging.herokuapp.com';
```
Comment (s-akinrele) [inline src/constants/Constants.js:1]: maybe move this to the .env file?

File: src/containers/Signup/Signup.js [hunk 4]
```diff
@@ -24,13 +25,19 @@ class Signup extends Component {
...
+    this.initialState = {
       email: '',
       username: '',
       password: '',
       confirmPassword: ''
     };
 
+    this.state = {
+      ...this.initialState
+    };
```
Comment (s-akinrele) [inline src/containers/Signup/Signup.js:39]: 😄 fancy but not need ... 
you can just add the values to the state object directly... 
benefits: removes unnecessary declaration and reduces the lines of code in this file

File: src/containers/Signup/Signup.js [hunk 5]
```diff
@@ -105,7 +112,10 @@ class Signup extends Component {
       return;
     }
 
-    this.props.signup(this.state);
+    this.props.signup(this.state)
+    .then((res) => {
+      this.setState({ ...this.initialState });
```
Comment (s-akinrele) [inline src/containers/Signup/Signup.js:117]: oh now I see.... okay to can ignore my comment above

Comment (kampkelly) [reply src/containers/Signup/Signup.js:117]: okay.

**Review by s-akinrele — APPROVED** [review]
GTM...

**Pull request merged**